File: src/main/java/com/rsupport/download/google/drive/FileExtension.java
package com.rsupport.download.google.drive;

import com.google.api.services.drive.model.File;

import java.util.HashMap;
import java.util.Map;

interface FileExtension {
    String extension();

    String exportMimeType();

    static FileExtension from(File file) {
        return new DriveExtension(file);
    }
}

class DriveExtension implements FileExtension {

    private final File file;

    DriveExtension(File file) {
        this.file = file;
    }

    private Map<String, String> extensionMap() {
        final Map<String, String> map = new HashMap<>();
        map.put("application/vnd.google-apps.spreadsheet", "xlsx");
        return map;
    }

    private Map<String, String> outMimeTypeMap() {
        final Map<String, String> map = new HashMap<>();
        map.put("application/vnd.google-apps.spreadsheet", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");
        return map;
    }

    @Override
    public String extension() {
        return extensionMap().get(file.getMimeType());
    }

    @Override
    public String exportMimeType() {
        return outMimeTypeMap().get(file.getMimeType());
    }
}

File: src/main/java/com/rsupport/download/google/drive/FileExtension.kt
package com.rsupport.download.google.drive

import com.google.api.services.drive.model.File

interface FileExtension{
    fun extension(): String
    fun exportMimeType(): String

    companion object{
        fun from(file: File): FileExtension {
            return DriveFileExtension(file)
        }
    }
}

class DriveFileExtension(private val file: File): FileExtension {
    private val extensionMap = mapOf(
            "application/vnd.google-apps.spreadsheet" to "xlsx"
    )

    private val outMimeTypeMap = mapOf(
            "application/vnd.google-apps.spreadsheet" to "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
    )

    override fun extension(): String {
        return extensionMap[file.mimeType] ?: ""
    }

    override fun exportMimeType(): String {
        return outMimeTypeMap[file.mimeType] ?: ""
    }
}